org.hl7.fhir.r5/src/test/java/org/hl7/fhir/r5/test/CanonicalResourceManagerTests.java

```diff
@@ -927,4 +929,877 @@ public void testSupplements() {
+    Assertions.assertEquals("1", csExact2_1_0.getId());
+
+    CodeSystem csExact2_1_0_alpha = csm.get("http://test", "2.1.0-alpha");
+    Assertions.assertEquals("2", csExact2_1_0_alpha.getId());
+
+    CodeSystem csPartial2_1 = csm.get("http://test", "2.1");
+    Assertions.assertEquals("1", csPartial2_1.getId()); // Should return release version
+
+    CodeSystem csLatest = csm.get("http://test", null);
+    Assertions.assertEquals("1", csLatest.getId());
+  }
+
+  @Test
+  public void testSemverDepth_2_1_0_alpha_Then_2_1_0() throws ParseException {
+    CanonicalResourceManager<CodeSystem> csm = new CanonicalResourceManager<CodeSystem>(false, false);
+
+    CodeSystem cs1 = makeCodeSystem("1", "http://test", "2.1.0-alpha");
+    CanonicalResourceProxy cp1 = new DeferredLoadTestResource(cs1);
+    PackageInformation pi1 = makePackageInfo("hl7.fhir.core", "1.0.0", "2023-01-01");
+    CanonicalResourceManager.CachedCanonicalResource cc1 = csm.new CachedCanonicalResource<CodeSystem>(cp1, pi1);
+    csm.see(cc1);
+
+    CodeSystem cs2 = makeCodeSystem("2", "http://test", "2.1.0");
+    CanonicalResourceProxy cp2 = new DeferredLoadTestResource(cs2);
+    PackageInformation pi2 = makePackageInfo("hl7.fhir.core", "1.0.0", "2023-01-01");
+    CanonicalResourceManager.CachedCanonicalResource cc2 = csm.new CachedCanonicalResource<CodeSystem>(cp2, pi2);
+    csm.see(cc2);
+
+    // 2.1.0 > 2.1.0-alpha (release > pre-release, regardless of order)
+    CodeSystem csExact2_1_0 = csm.get("http://test", "2.1.0");
+    Assertions.assertEquals("2", csExact2_1_0.getId());
+
+    CodeSystem csExact2_1_0_alpha = csm.get("http://test", "2.1.0-alpha");
+    Assertions.assertEquals("1", csExact2_1_0_alpha.getId());
+
+    CodeSystem csPartial2_1 = csm.get("http://test", "2.1");
+    Assertions.assertEquals("2", csPartial2_1.getId()); // Should return release version
+
+    CodeSystem csLatest = csm.get("http://test", null);
+    Assertions.assertEquals("2", csLatest.getId());
+  }
+
+  @Test
+  public void testSemverDepth_2_1_Then_2_1_0_alpha() throws ParseException {
+    CanonicalResourceManager<CodeSystem> csm = new CanonicalResourceManager<CodeSystem>(false, false);
+
+    CodeSystem cs1 = makeCodeSystem("1", "http://test", "2.1");
+    CanonicalResourceProxy cp1 = new DeferredLoadTestResource(cs1);
+    PackageInformation pi1 = makePackageInfo("hl7.fhir.core", "1.0.0", "2023-01-01");
+    CanonicalResourceManager.CachedCanonicalResource cc1 = csm.new CachedCanonicalResource<CodeSystem>(cp1, pi1);
+    csm.see(cc1);
+
+    CodeSystem cs2 = makeCodeSystem("2", "http://test", "2.1.0-alpha");
+    CanonicalResourceProxy cp2 = new DeferredLoadTestResource(cs2);
+    PackageInformation pi2 = makePackageInfo("hl7.fhir.core", "1.0.0", "2023-01-01");
+    CanonicalResourceManager.CachedCanonicalResource cc2 = csm.new CachedCanonicalResource<CodeSystem>(cp2, pi2);
+    csm.see(cc2);
+
+    // 2.1.0-alpha > 2.1 (more specific wins)

```

Comment:
This doesn't feel right. Doesn't 2.1 constitute an actual release (resolving to 2.1.0) and 2.1.0-alpha a pre-release?